rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/HostsFileBasedInfrastructureManager.java

```diff
@@ -627,4 +545,62 @@ public Void handle() {
         });
     }
 
+    /**
+     * Utility class to encapsulate node name building and node name parsing.
+     * The node name involves the host that was written in the host file, and
+     * that si used as an identifying key to track the node afterwards.
+     *
+     * Configured host addresses need to be converted to compliant host
+     * addresses in terms of node name building rules (e.g. prevent dots in
+     * node names)
+     */
+    protected class NodeNameBuilder implements Serializable {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this inner class static [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2066)